trustmanager/keyfilestore.go

```diff
@@ -24,11 +40,73 @@ func NewKeyFileStore(baseDir string) (*KeyFileStore, error) {
 		return nil, err
 	}
 
-	return &KeyFileStore{fileStore}, nil
+	return &KeyFileStore{*fileStore}, nil
 }
 
 // AddKey stores the contents of a PEM-encoded private key as a PEM block
 func (s *KeyFileStore) AddKey(name string, privKey *data.PrivateKey) error {
+	return addKey(s, name, privKey)
+}
+
+// GetKey returns the PrivateKey given a KeyID
+func (s *KeyFileStore) GetKey(name string) (*data.PrivateKey, error) {
+	return getKey(s, name)
+}
+
+// AddEncryptedKey stores the contents of a PEM-encoded private key as an encrypted PEM block
+func (s *KeyFileStore) AddEncryptedKey(name string, privKey *data.PrivateKey, passphrase string) error {
+	return addEncryptedKey(s, name, privKey, passphrase)
+}
+
+// GetDecryptedKey decrypts and returns the PEM Encoded private key given a flename
+// and a passphrase
+func (s *KeyFileStore) GetDecryptedKey(name string, passphrase string) (*data.PrivateKey, error) {
+	return getDecryptedKey(s, name, passphrase)
+}
+
+// ListKeys returns a list of unique PublicKeys present on the KeyFileStore.
+// There might be symlinks associating Certificate IDs to Public Keys, so this
+// method only returns the IDs that aren't symlinks
+func (s *KeyFileStore) ListKeys() []string {
+	return listKeys(s)
+}
+
+// NewKeyMemoryStore returns a new KeyMemoryStore which holds keys in memory
+func NewKeyMemoryStore() *KeyMemoryStore {
```

Comment:
I don't see any tests for this new MemoryFileStore. Did I missed them somewhere?